zerver/migrations/0680_rename_general_chat_to_empty_string_topic.py

```diff
@@ -0,0 +1,119 @@
+    if max_id is None:
+        return
+
+    while lower_id_bound < max_id:
+        upper_id_bound = min(lower_id_bound + BATCH_SIZE, max_id)
+        with connection.cursor() as cursor:
+            query = SQL(
+                """
+                UPDATE {table_name}
+                SET edit_history = (
+                    SELECT JSONB_AGG(
+                        elem
+                        ||
+                        (CASE
+                            WHEN elem ? 'prev_topic' AND elem->>'prev_topic' = 'general chat'
+                            THEN '{{"prev_topic": ""}}'::jsonb
+                            ELSE '{{}}'::jsonb
+                        END)
+                        ||
+                        (CASE
+                            WHEN elem ? 'topic' AND elem->>'topic' = 'general chat'
+                            THEN '{{"topic": ""}}'::jsonb
+                            ELSE '{{}}'::jsonb
+                        END)
+                    )::text
+                    FROM JSONB_ARRAY_ELEMENTS(edit_history::jsonb) AS elem
+                )
+                WHERE edit_history IS NOT NULL
+                AND id > %(lower_id_bound)s AND id <= %(upper_id_bound)s
+                AND (
+                    edit_history::jsonb @> '[{{"prev_topic": "general chat"}}]' OR
+                    edit_history::jsonb @> '[{{"topic": "general chat"}}]'
+                );
+            """
+            ).format(table_name=Identifier(message_model._meta.db_table))
+            cursor.execute(
+                query,
+                {
+                    "lower_id_bound": lower_id_bound,
+                    "upper_id_bound": upper_id_bound,
+                },
+            )
+
+        print(f"Processed {upper_id_bound} / {max_id}")
+        lower_id_bound += BATCH_SIZE
+
+
+def rename_general_chat_to_empty_string_topic(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    Realm = apps.get_model("zerver", "Realm")
+    Message = apps.get_model("zerver", "Message")
+    UserTopic = apps.get_model("zerver", "UserTopic")
+    ArchivedMessage = apps.get_model("zerver", "ArchivedMessage")
+    ScheduledMessage = apps.get_model("zerver", "ScheduledMessage")
+
+    for realm in Realm.objects.all():
+        with transaction.atomic(durable=True):
+            # Uses index "zerver_message_realm_upper_subject"
+            message_queryset = Message.objects.filter(realm=realm, subject__iexact="general chat")
+            channel_ids = list(
+                message_queryset.distinct("recipient__type_id").values_list(
+                    "recipient__type_id", flat=True
+                )
+            )
+
+            message_queryset.update(subject="")
+
+            # Uses index "zerver_mutedtopic_stream_topic"
+            UserTopic.objects.filter(
+                stream_id__in=channel_ids, topic_name__iexact="general chat"
+            ).update(topic_name="")
```

Comment:
This is not guaranteed to be complete, in that in theory you could have a muted record for a topic that doesn't exist. But I think it's OK to not do anything with those -- structurally, they're inoperative.

Reply:
I think the actual behavior is fine; will add a comment for posterity.